saleor/order/actions.py

```diff
@@ -297,6 +297,10 @@ def order_created(
             extra={"tax_error": order.tax_error, "order_id": order_id},
         )
 
+    if order_user := order.user:
+        order_user.number_of_orders += 1
```

Comment:
I would recommend to use F expression. There is small possibility of overwriting the data, but still IMHO worth to have it:
```python
order_user.number_of_orders = F("number_of_orders") + 1
```